pydantic_ai_slim/pydantic_ai/durable_exec/dbos/_fastmcp_toolset.py

```diff
@@ -0,0 +1,100 @@
+from __future__ import annotations
+
+from abc import ABC
+from collections.abc import Callable
+from typing import TYPE_CHECKING, Any
+
+from dbos import DBOS
+from typing_extensions import Self
+
+from pydantic_ai import AbstractToolset, ToolsetTool, WrapperToolset
+from pydantic_ai.tools import AgentDepsT, RunContext, ToolDefinition
+from pydantic_ai.toolsets.fastmcp import FastMCPToolset
+
+from ._utils import StepConfig
+
+if TYPE_CHECKING:
+    from pydantic_ai.mcp import ToolResult
+
+
+class DBOSFastMCPToolset(WrapperToolset[AgentDepsT], ABC):
```

Comment:
Would it be an option to follow the same approach we took with Temporal, where both the MCPServer and FastMCPToolset wrappers inherit from a common MCP wrapper?